src/Avalonia.Controls.TreeDataGrid/TreeDataGrid.cs

```diff
@@ -239,6 +240,11 @@ protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
             Scroll = e.NameScope.Find<ScrollViewer>("PART_ScrollViewer");
         }
 
+        protected void OnPreviewKeyDown(object? o, KeyEventArgs e)
```

Comment:
Because ScrollViewer handles OnKeyDown even before i receive it and as i understand that's the only option left
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ScrollViewer.cs#L734-L743

Reply:
I am not sure what you mean, if I will try to override OnKeyDown in TreeDataGridRowsPresenter it still wouldn't be fired when I click PgUp\PgDown. In case I would try to handle PreviewOnKeyDown here I am not sure how it would help.